### PR TITLE
refactor qwenimage pipeline tests to the new mixin structure

### PR DESCRIPTION
**Pull request by akshan-main**
# What does this PR do?

Refactors the qwenimage pipeline tests to the new mixin structure from #14113: a `QwenImagePipelineTesterConfig` holding the components and inputs, `TestQwenImagePipeline(PipelineTesterMixin)` for the fast tests, and `TestQwenImagePipelineMemory(MemoryTesterMixin)` for the offload and casting tests.

Existing coverage is kept as-is: `test_inference` and the qwen-specific tests (VAE tiling, true-CFG without a negative mask). No new tests are introduced since the old suite had no caching tests. The attention-slicing test is dropped since the new framework has no equivalent.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Discussed on Slack with @sayakpaul
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

## Who can review?

@sayakpaul

### CODE REVIEW

File: tests/pipelines/qwenimage/test_qwenimage.py [hunk 1]
```diff
@@ -115,134 +101,62 @@ def get_dummy_components(self):
...
             "height": 32,
             "width": 32,
             "max_sequence_length": 16,
+            # Request torch outputs so tests compare torch tensors directly (see `BasePipelineTesterConfig`).
             "output_type": "pt",
         }
 
-        return inputs
-
-    def test_inference(self):
```
Comment (sayakpaul) [inline tests/pipelines/qwenimage/test_qwenimage.py:148]: Let's keep this test perhaps in `TestQwenImagePipeline`?

File: tests/pipelines/qwenimage/test_qwenimage.py [hunk 2]
```diff
@@ -258,4 +172,26 @@ def test_true_cfg_without_negative_prompt_embeds_mask(self):
...
+class TestQwenImagePipelineFirstBlockCache(QwenImagePipelineTesterConfig, FirstBlockCacheTesterMixin):
+    pass
+
+
+class TestQwenImagePipelineTaylorSeerCache(QwenImagePipelineTesterConfig, TaylorSeerCacheTesterMixin):
+    pass
+
+
+class TestQwenImagePipelineMagCache(QwenImagePipelineTesterConfig, MagCacheTesterMixin):
+    pass
```
Comment (sayakpaul) [inline tests/pipelines/qwenimage/test_qwenimage.py]: If the QwenImage test suite didn't have caching tests, let's not introduce them here.

**Review by sayakpaul — COMMENTED** [review]
Thanks for getting started so quickly!

File: tests/pipelines/qwenimage/test_qwenimage.py (hunk 2 above)
Comment (akshan-main) [reply tests/pipelines/qwenimage/test_qwenimage.py]: removed. nothing new added

File: tests/pipelines/qwenimage/test_qwenimage.py [hunk 3]
```diff
@@ -115,134 +97,79 @@ def get_dummy_components(self):
...
 
         generated_slice = generated_image.flatten()
         generated_slice = torch.cat([generated_slice[:8], generated_slice[-8:]])
-        self.assertTrue(torch.allclose(generated_slice, expected_slice, atol=5e-3))
+        assert torch.allclose(generated_slice, expected_slice, atol=5e-3)
 
     def test_inference_batch_single_identical(self):
-        self._test_inference_batch_single_identical(batch_size=3, expected_max_diff=1e-1)
-
-    def test_attention_slicing_forward_pass(
```
Comment (sayakpaul) [inline tests/pipelines/qwenimage/test_qwenimage.py:172]: Is this test already covered?

**Review by sayakpaul — APPROVED** [review]

File: tests/pipelines/qwenimage/test_qwenimage.py (hunk 3 above)
Comment (akshan-main) [reply tests/pipelines/qwenimage/test_qwenimage.py:172]: Yes, it's inherited from the base PipelineTesterMixin. The override only loosens the tolerance to 1e-1 (Qwen's batched vs single output exceeds the 1e-4 default, the same value the old test used).

Comment (akshan-main) [reply tests/pipelines/qwenimage/test_qwenimage.py:172]: Dropped it. Passes at the base 1e-4 default, so the override wasn't needed.

**Review by sayakpaul — APPROVED** [review]
